### ISSUE TITLE
Dired async should have a progress bar for remote operations

### ISSUE THREAD

**Issue by monsanto** [jwiegley/emacs-async#4]
See title. I think Emacs wget does a great job of this (very hard to find elisp, archived here https://github.com/monsanto/wget). A simple way to do this would be to add native commands for remote copying using scp, and associate the output of that command with a buffer.

**Comment by azzamsa** [comment]
Yes. [long time ago](https://www.reddit.com/r/emacs/comments/7fdn4o/how_to_make_progress_bar_in_dired/) I also think about this. Maybe using [spinner](https://github.com/Malabarba/spinner.el). But I don't have any clue if this doable in dired.

**Comment by Yevgnen** [comment]
At least if it has some percentage information, that would be very nice.

**Comment by thierryvolpiatto** [comment]
It would be nice indeed, but I don't know how to do it reliably. I have a branch that provide a progress reporter with percentage in the mode-line, but it works only for local files, I think I use a wrong approach by checking the amount transferred on destination, perhaps somebody can enlighten me on how to proceed. The best would be to get the speed transfer from kernel (/proc/?) every n seconds and deduce the percentage with this information (on other platforms ???).
I can push the branch here if somebody interested.

**Comment by thierryvolpiatto** [comment]
Done, pushed as progress_reporter.

![Capture d’écran_2019-03-23_08-25-06](https://user-images.githubusercontent.com/1533939/54863161-a18d8780-4d45-11e9-8416-10f184f643a2.png)
![Capture d’écran_2019-03-23_08-25-23](https://user-images.githubusercontent.com/1533939/54863168-b407c100-4d45-11e9-8ebd-7b2b011bde60.png)

**Comment by fu123456** [comment]
@thierryvolpiatto, you said "Done, pushed as progress_reporter.". Can you share your code? I cannot find any code about this. Thx.

**Comment by thierryvolpiatto** [comment]
https://github.com/jwiegley/emacs-async/tree/progress_reporter

Not sure if it is still working, was working more or less but was buggy anyway.

**Comment by dinojr** [comment]
The link you provided is dead. Is it available somewhere else ?

**Comment by thierryvolpiatto** [comment]
Yes, I think I definitely lost this branch.
